gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -107,6 +120,7 @@ public final class InstantiatingGrpcChannelProvider implements TransportChannelP
   private final int processorCount;
   private final Executor executor;
   private final HeaderProvider headerProvider;
+  private final EndpointContext endpointContext;
   private final String endpoint;
```

Comment:
it seems a bit confusing to have both endpoint and endpointContext co-exist here. I wonder if it is possible to either remove `endpoint`, or derive it from `endpointContext` when makes sense?
iiuc, endpoint is only used in ClientContext, but it is a public method from parent class https://github.com/googleapis/sdk-platform-java/blob/ca3ec24f506f0d43623a92ba0eed908f874fe488/gax-java/gax/src/main/java/com/google/api/gax/rpc/ClientContext.java#L223
and value is resolved from 
https://github.com/googleapis/sdk-platform-java/blob/ca3ec24f506f0d43623a92ba0eed908f874fe488/gax-java/gax/src/main/java/com/google/api/gax/rpc/ClientContext.java#L191
Maybe we can remove usages within gax and mark it as deprecated.
might worth consider refactoring? WDYT @lqiu96

Reply:
Yeah agreed. I think we can remove the internal usages of endpoint inside IntantiatingGrpcChannelProvider and IntantiatingHttpJsonChannelProvider and use `endpointContext.resolvedEndpoint()`. I don't think we can mark `withEndpoint()` as deprecated as we treat adding the deprecated annotation as a breaking change in our client libraries. 

I think this refactor is related to the PR, but can also be done separately as this PR add in any new public methods for endpoint. If you want to take a stab at it Riya, please feel free to do it. Otherwise, I think Min or I can do the refactor and add @ObsoleteApi to it.

WDYT @zhumin8

Reply:
I addressed in https://github.com/googleapis/sdk-platform-java/pull/3326/commits/926facae0f29ef653a75c63a793908440cf4b3e3. Let me know what you both think